core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala

```diff
@@ -2062,6 +2067,86 @@ class PlaintextAdminIntegrationTest extends BaseAdminIntegrationTest {
     }
   }
 
+  /**
+   * Test the consumer group APIs.
+   */
+  @ParameterizedTest
+  @ValueSource(strings = Array("kraft"))
+  def testConsumerGroupWithMemberMigration(quorum: String): Unit = {
+    val config = createConfig
+    client = Admin.create(config)
+    var classicConsumer: Consumer[Array[Byte], Array[Byte]] = null
+    var consumerConsumer: Consumer[Array[Byte], Array[Byte]] = null
+    try {
+      // Verify that initially there are no consumer groups to list.
+      val list1 = client.listConsumerGroups
+      assertEquals(0, list1.all.get.size)
+      assertEquals(0, list1.errors.get.size)
+      assertEquals(0, list1.valid.get.size)
+      val testTopicName = "test_topic"
+      val testNumPartitions = 2
+
+      client.createTopics(util.Arrays.asList(
+        new NewTopic(testTopicName, testNumPartitions, 1.toShort),
+      )).all.get
+      waitForTopics(client, List(testTopicName), List())
+
+      val producer = createProducer()
+      try {
+        producer.send(new ProducerRecord(testTopicName, 0, null, null))
+        producer.send(new ProducerRecord(testTopicName, 1, null, null))
+        producer.flush()
+      } finally {
+        Utils.closeQuietly(producer, "producer")
+      }
+
+      val testGroupId = "test_group_id"
+      val testClassicClientId = "test_classic_client_id"
+      val testConsumerClientId = "test_consumer_client_id"
+
+      val newConsumerConfig = new Properties(consumerConfig)
+      newConsumerConfig.put(ConsumerConfig.GROUP_ID_CONFIG, testGroupId)
+      newConsumerConfig.put(ConsumerConfig.CLIENT_ID_CONFIG, testClassicClientId)
+      consumerConfig.put(ConsumerConfig.GROUP_PROTOCOL_CONFIG, GroupProtocol.CLASSIC.name)
+
+      classicConsumer = createConsumer(configOverrides = newConsumerConfig)
+      classicConsumer.subscribe(List(testTopicName).asJava)
+      classicConsumer.poll(JDuration.ofMillis(1000))
+
+      newConsumerConfig.put(ConsumerConfig.CLIENT_ID_CONFIG, testConsumerClientId)
+      consumerConfig.put(ConsumerConfig.GROUP_PROTOCOL_CONFIG, GroupProtocol.CONSUMER.name)
+      consumerConsumer = createConsumer(configOverrides = newConsumerConfig)
+      consumerConsumer.subscribe(List(testTopicName).asJava)
+      consumerConsumer.poll(JDuration.ofMillis(1000))
+
+      TestUtils.waitUntilTrue(() => {
+        val describeConsumerGroupResult = client.describeConsumerGroups(Seq(testGroupId).asJava).all.get
+        describeConsumerGroupResult.containsKey(testGroupId) &&
+          describeConsumerGroupResult.get(testGroupId).members.size == 2
```

Comment:
Here you could also wait until the group becomes STABLE. It will ease the assertions afterwards.

Reply:
Thanks for the suggestion. I change to wait until group becomes RECONCILING, because the group is during migration.

Reply:
Why RECONCILING? The group should eventually switch to STABLE anyway. Could you please elaborate?

Reply:
Sorry, my mistake. I add `classicConsumer#poll` and `consumerConsumer#poll` to `TestUtils#waitUntilTure` function. Finally, the group can be STABLE. Thanks.